app/main/views/unsubscribe_requests.py

```diff
@@ -13,16 +13,27 @@ def unsubscribe_request_reports_summary(service_id):
 
 
 @main.route("/services/<uuid:service_id>/unsubscribe-requests/reports/latest")
-@main.route("/services/<uuid:service_id>/unsubscribe-requests/reports/<uuid:batch_id>")
+@main.route("/services/<uuid:service_id>/unsubscribe-requests/reports/<uuid:batch_id>", methods=["GET", "POST"])
 @user_has_permissions("view_activity")
 def unsubscribe_request_report(service_id, batch_id=None):
     if batch_id:
         report = current_service.unsubscribe_request_reports_summary.get_by_batch_id(batch_id)
     else:
         report = current_service.unsubscribe_request_reports_summary.get_unbatched_report()
-    form = ProcessUnsubscribeRequestForm(is_a_batched_report=report.is_a_batched_report)
+    form = ProcessUnsubscribeRequestForm(is_a_batched_report=report.is_a_batched_report, report_status=report.status)
```

Comment:
I think you can use WTForms’ built-in logic here to set the checkbox as checked/not checked (by making the argument name the same as the name of the field instance):
```suggestion
    form = ProcessUnsubscribeRequestForm(is_a_batched_report=report.is_a_batched_report, report_has_been_processed=(report.status == "Completed"))
```

You’d then be able to remove a lot of the bespoke code from the form’s `__init__` method.